lib/data_update_scripts/20200916202343_backfill_co_author_ids_for_articles.rb

```diff
@@ -0,0 +1,10 @@
+module DataUpdateScripts
+  class BackfillCoAuthorIdsForArticles
+    def run
+      articles = Article.where.not(second_user_id: nil).or(Article.where.not(third_user_id: nil))
+      articles.find_each do |article|
```

Comment:
Here we can update all articles in a single SQL statement, without instantiating any `ActiveRecord` objects:

```ruby
articles.update_all("co_author_ids = array[second_user_id, third_user_id]")
```

This will issue the following statement:

```sql
UPDATE "articles" 
SET co_author_ids = array[second_user_id, third_user_id] 
WHERE ("articles"."second_user_id" IS NOT NULL OR "articles"."third_user_id" IS NOT NULL);
```

I think we should let the database do the heavy lifting here since we don't actually need `ActiveRecord` instances for this.

Reply:
This is an **excellent** suggestion and I've since made this change @citizen428!